### PR TITLE
Adopt provider metadata issuer from discovery document when doing token verification

### PR DESCRIPTION
**Pull request by eJamesLin**
### About
- When verifying token, get the issuer metadata from open ID discovery document, instead of a fixed value.
- https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata

### CODE REVIEW

File: LineSDK/LineSDK/Login/LoginManager.swift [hunk 1]
```diff
@@ -288,7 +290,7 @@ extension LoginManager {
     func getJWK(
```
Comment (onevcat) [inline LineSDK/LineSDK/Login/LoginManager.swift]: Let's rename this method to something like `getProviderMetadata`

Comment (eJamesLin) [reply LineSDK/LineSDK/Login/LoginManager.swift]: indeed. got it~ thanks~

File: LineSDK/LineSDK/Login/LoginManager.swift [hunk 2]
```diff
@@ -334,12 +336,12 @@ extension LoginManager {
         }
     }
     
-    func verifyIDToken(_ token: JWT, key: JWK, process: LoginProcess, userID: String?) throws {
+    func verifyIDToken(_ token: JWT, expected: DiscoveryDocument.ResolvedProviderMetadata, process: LoginProcess, userID: String?) throws {
```
Comment (onevcat) [inline LineSDK/LineSDK/Login/LoginManager.swift]: I prefer some new line to make the code not extending too much. Can you set the "Page Guide" at column 120?

Comment (eJamesLin) [reply LineSDK/LineSDK/Login/LoginManager.swift]: oh, thanks for the remind. My previous `page guide` seems gone after switching between Xcode 10/11 😂

File: LineSDK/LineSDK/Login/LoginManager.swift [hunk 3]
```diff
@@ -187,7 +187,9 @@ public class LoginManager {
         let group = DispatchGroup()
         
         var profile: UserProfile?
-        var webToken: JWK?
+
+        var expected: DiscoveryDocument.ResolvedProviderMetadata?
```
Comment (onevcat) [inline LineSDK/LineSDK/Login/LoginManager.swift]: I suggest `providerMetadata` a better name. The `issuer` is the expected value, but the `JWK` is not what is expected, it is just a key to be used while verification.

**Comment by onevcat** [comment]
It's nice!

**Pull request merged**